File: blockchain/blockchain.go
package blockchain

import (
	"crypto/sha256"
	"encoding/hex"
	"time"
)

// Block represents a block in the blockchain
type Block struct {
	Index        int
	Timestamp    time.Time
	PreviousHash string
	Hash         string
	Data         string
}

// Blockchain represents the blockchain
type Blockchain struct {
	Chain []Block
}

// NewBlock creates a new block in the blockchain
func (bc *Blockchain) NewBlock(data string, previousHash string) *Block {
	block := &Block{
		Index:        len(bc.Chain),
		Timestamp:    time.Now(),
		PreviousHash: previousHash,
		Data:         data,
	}
	block.Hash = block.calculateHash()
	return block
}

// AddBlock adds a new block to the blockchain
func (bc *Blockchain) AddBlock(data string) {
	previousBlock := bc.Chain[len(bc.Chain)-1]
	newBlock := bc.NewBlock(data, previousBlock.Hash)
	bc.Chain = append(bc.Chain, *newBlock)
}

// calculateHash calculates the hash of a block
func (b *Block) calculateHash() string {
	record := string(b.Index) + b.Timestamp.String() + b.PreviousHash + b.Data
	h := sha256.New()
	h.Write([]byte(record))
	hashed := h.Sum(nil)
	return hex.EncodeToString(hashed)
}

// NewBlockchain creates a new blockchain with the genesis block
func NewBlockchain() *Blockchain {
	genesisBlock := &Block{
		Index:        0,
		Timestamp:    time.Now(),
		PreviousHash: "",
		Data:         "Genesis Block",
	}
	genesisBlock.Hash = genesisBlock.calculateHash()
	return &Blockchain{Chain: []Block{*genesisBlock}}
}

File: blockchain/blockchain.py
import hashlib
import datetime

class Block:
    def __init__(self, index, timestamp, previous_hash, data):
        self.index = index
        self.timestamp = timestamp
        self.previous_hash = previous_hash
        self.data = data
        self.hash = self.calculate_hash()

    def calculate_hash(self):
        sha = hashlib.sha256()
        sha.update(str(self.index).encode('utf-8') +
                    str(self.timestamp).encode('utf-8') +
                    str(self.previous_hash).encode('utf-8') +
                    str(self.data).encode('utf-8'))
        return sha.hexdigest()

class Blockchain:
    def __init__(self):
        self.chain = [self.create_genesis_block()]

    def create_genesis_block(self):
        return Block(0, datetime.datetime.now(), "0", "Genesis Block")

    def add_block(self, data):
        previous_block = self.chain[-1]
        new_block = Block(len(self.chain), datetime.datetime.now(), previous_block.hash, data)
        self.chain.append(new_block)
